### DISCUSSION TITLE
Boundary Generation

### DISCUSSION THREAD

**Discussion by styyokuda** [idaholab/moose#25903] [Q&A Meshing] [answered]
All,
I would like to generate boundaries. I used to do like:
```
[MeshModifiers]
  [./inner_BC]
  type = ParsedAddSideset
  combinatorial_geometry = 'x < 0.001'
  new_sideset_name = 'inner_boundary'
  [../]
[ ]
```
But, this method seems not to work any longer.
Please let me know how to generate boundaries, on which I can apply boundary conditions.
S. Thomas

**Comment by GiudGiud** [comment]
MeshModifiers have been retired years ago!
We now use MeshGenerators. 

For this one:
```
[Mesh]
  [./inner_BC]
  type = ParsedSidesetGenerator
  combinatorial_geometry = 'x < 0.001'
  new_sideset_name = 'inner_boundary'
  [../]
[ ]
```

**Reply by styyokuda** [reply]
Thank you, Guillaume.

I got the following error messages:
```
*** ERROR ***
~/mesh_generator.i:57: section '[MeshGenerators]' does not have an associated "Action".
 Common causes:
- you misspelled the Action/section name
- the app you are running does not support this Action/syntax
```
Please help.
S. Thomas

**Reply by GiudGiud** [reply]
oh my bad
[Mesh]

**Reply by styyokuda** [reply]
Thank you, Guillaume.
It seems to be "ParsedGeneratorSideset" instead of "ParsedSidesetGenerator" :)
"ParsedGeneratorSideset" works ... thank you.

Is there anyway to delete the old boundaries?
or keeping old boundaries, which are not used any longer, doesn't harm the computation?
S. Thomas

**Reply by GiudGiud** [reply]
there is this one
https://mooseframework.inl.gov/source/meshgenerators/BoundaryDeletionGenerator.html

but having old boundaries does not harm anything

**Reply by styyokuda** [reply, marked as answer]
Thank you, Guillaume, as always. :)
